controller/src/main/java/gov/va/vro/controller/advice/InputSanitizerAdvice.java

```diff
@@ -0,0 +1,94 @@
+package gov.va.vro.controller.advice;
+
+import gov.va.vro.controller.exception.DisallowedPatternException;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpInputMessage;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.servlet.mvc.method.annotation.RequestBodyAdvice;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@ControllerAdvice
+@ComponentScan
+public class InputSanitizerAdvice implements RequestBodyAdvice {
+
+  @Override
+  public boolean supports(
+      MethodParameter methodParameter,
+      Type targetType,
+      Class<? extends HttpMessageConverter<?>> converterType) {
+    // Return true to indicate that this advice should be applied to all request
+    // bodies
+    return true;
+  }
+
+  @Override
+  public HttpInputMessage beforeBodyRead(
+      HttpInputMessage inputMessage,
+      MethodParameter parameter,
+      Type targetType,
+      Class<? extends HttpMessageConverter<?>> converterType)
+      throws IOException {
+    // Create a custom HttpInputMessage that sanitizes the request body before it is
+    // read
+    return new HttpInputMessage() {
+      @Override
+      public InputStream getBody() throws IOException {
+        // Read the request body and remove non-printable and NUL characters from it
+        String requestBody =
+            new String(inputMessage.getBody().readAllBytes(), StandardCharsets.UTF_8);
+
+        // https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html
+        Pattern nonPrintablePattern = Pattern.compile("\\P{Print}&&[^{}]");
```

Comment:
Why are these not static? Compiles are expensive as far as I know. I looked and they seem to be thread safe as well.

Reply:
So any Spring bean is a singleton. And `Pattern.compile()` is only done once, and is best if you need to reuse the pattern a bunch of times. And since this is a pointcut @Advice method, this will be used a bunch of times (anytime we see a `RequestBody`). I don't think static will do any good in this situation.

Reply:
@wlin2019 and @dimitri-amida can fact-check me on this, tho.

Reply:
Spoke to Afsin and he is correct with how I am declaring it here being non-performant. It should be a class-level field. For static variables in Singletons, it doesn't matter, but I will declare it static anyway to be prudent.